### PR TITLE
feat(create article): UA event added for createArticle

### PR DESCRIPTION
**Pull request by SimonMilord**
[SFINT-5276](https://coveord.atlassian.net/browse/SFINT-5276)

**The following custom event has been added to Insight Client for knowledge article creation:**

- createArticle

[SFINT-5276]: https://coveord.atlassian.net/browse/SFINT-5276?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/searchPage/searchPageEvents.ts [hunk 1]
```diff
@@ -389,6 +389,7 @@ export const CustomEventsTypes: Partial<Record<SearchPageEvents | InsightEvents,
     [SearchPageEvents.generatedAnswerHideAnswers]: 'generatedAnswer',
     [SearchPageEvents.generatedAnswerShowAnswers]: 'generatedAnswer',
     [SearchPageEvents.generatedAnswerFeedbackSubmit]: 'generatedAnswer',
+    [InsightEvents.createArticle]: 'createArticle',
```
Comment (erocheleau) [inline src/searchPage/searchPageEvents.ts:392]: Is there any kind of order here? I would group it with the other InsightEvents :D

**Review by erocheleau — APPROVED** [review]
looks good

File: src/searchPage/searchPageEvents.ts (hunk 1 above)
Comment (SimonMilord) [reply src/searchPage/searchPageEvents.ts:392]: I thought newer events at the bottom, but I can move it 😂

**Review by mmitiche — CHANGES_REQUESTED** [review]
In the Create Article event currently sent in the standard agent panel, the event contains the following custom data:

Screenshot 2023-11-22 at 7 08 12 AM

Notice the fields `articleType` and `triggeredBy` are being sent in addition to the context.

Please consider adding these two fields to the metadata accepted by the `logCreateArticle` function added in this PR.

**Comment by erocheleau** [comment]
> In the Create Article event currently sent in the standard agent panel, the event contains the following custom data:
> 
> Screenshot 2023-11-22 at 7 08 12 AM
> Notice the fields `articleType` and `triggeredBy` are being sent in addition to the context.
> 
> Please consider adding these two fields to the metadata accepted by the `logCreateArticle` function added in this PR.

I would have to go verify, but I don't think we know the article type, do we? xD 
We used to have an option to specify the article type but we won't be able to have that in the HIP at least not yet.

File: src/insight/insightClient.ts [hunk 2]
```diff
@@ -266,6 +267,13 @@ export class CoveoInsightClient {
         });
     }
 
+    public logCreateArticle(createArticle: CreateArticleMetadata, metadata?: CaseMetadata) {
```
Comment (mmitiche) [inline src/insight/insightClient.ts:270]:
```suggestion
    public logCreateArticle(createArticleMetadata: CreateArticleMetadata, metadata?: CaseMetadata) {
```

File: src/insight/insightClient.ts [hunk 3]
```diff
@@ -266,6 +267,13 @@ export class CoveoInsightClient {
...
+        return this.logCustomEvent(
+            InsightEvents.createArticle,
+            metadata ? {...generateMetadataToSend(metadata, false), ...createArticle} : {createArticle}
```
Comment (mmitiche) [inline src/insight/insightClient.ts:273]: `createArticleMetadata` is already an object, we would get a different structure in the event custom data if we keep it `{createArticle}`
```suggestion
            metadata ? {...generateMetadataToSend(metadata, false), ...createArticle} : createArticle
```

File: src/insight/insightClient.ts (hunk 2 above)
Comment (SimonMilord) [reply src/insight/insightClient.ts:270]: I was using the same pattern as :

`logPageViewClick(pageView: UserActionsPageViewMetadata, metadata: CaseMetadata)`

but sure I can change it

**Review by mmitiche — APPROVED** [review]
LGTM 👌

**Pull request merged**